src/routes/api/admin/index.js

```diff
@@ -0,0 +1,74 @@
+const { Router } = require("express");
+const { auth } = require("@utils/discordApi");
+const Bots = require("@models/bots");
+const { MessageEmbed } = require('discord.js');
+const { server } = require("@root/config.json")
+const route = Router();
+
+route.patch("/:id", auth, async function (req, res) {
+    let data = req.body;
+    const bot = await Bots.findOne({ "state": "unverified", botid: req.params.id }, { _id: false });
+    if (!bot) return res.json({ success: false, message: 'Bot not found' })
+    let modLog = await req.app.get('client').channels.cache.get(server.mod_log_id);
+    let mod = await req.app.get('client').users.cache.get(req.user.id);
+    let message = await req.app.get('client').guilds.cache.get(server.id);
+    let botUser = await req.app.get('client').users.fetch(req.params.id);
+    const member = await req.app.get('client').guilds.cache.get(server.id).members.fetch(req.user.id);
+    if (!admin_user_ids.includes(req.user.id) && !member.roles.cache.has(bot_verifier)) return res.json({ success: false, message: 'Invalid User' });
+    if (data.method === 'approve') {
+        await Bots.updateOne({ botid: req.params.id }, { $set: { state: "verified", logo: botUser.displayAvatarURL({ format: "png", size: 256 }) } });
+        let owners = [bot.owners.primary].concat(bot.owners.additional);
+        let e = new MessageEmbed()
+            .setTitle('Bot Verified')
+            .addField(`Bot`, `<@${bot.botid}>`, true)
+            .addField(`Owner(s)`, owners.map(x => x ? `<@${x}>` : ""), true)
+            .addField("Mod", mod, true)
+            .setThumbnail(botUser.displayAvatarURL({ format: "png", size: 256 }))
+            .setTimestamp()
+            .setColor(0x26ff00);
+        modLog.send(e);
+        modLog.send(owners.map(x => x ? `<@${x}>` : "")).then(m => { m.delete(); });
+
+        owners = await message.members.fetch({ user: owners });
+        owners.forEach(o => {
+            o.roles.add(message.roles.cache.get(server.role_ids.bot_developer));
+            o.send(`Your bot \`${bot.username}\` has been verified.`);
+        });
+        message.members.fetch(message.client.users.cache.find(u => u.id === bot.botid)).then(bot => {
+            bot.roles.set([server.role_ids.bot, server.role_ids.verified]);
+        });
+        return res.json({ success: true })
+
+    } else if (data.method === 'deny') {
+        await Bots.updateOne({ botid: req.params.id }, { $set: { state: "deleted", logo: botUser.displayAvatarURL({ format: "png", size: 256 }) } });
+        let owners = [bot.owners.primary].concat(bot.owners.additional)
+        e = new MessageEmbed()
```

Comment:
You have forget 'let' before e = new MessageEmbed()

Reply:
I'm going to rewrite that file in multiple files soon with different endpoints for different actions.

Reply:
ok

Reply:
Huh My Bad 😞

Reply:
> I'm going to rewrite that file in multiple files soon with different endpoints for different actions.

Why ? Is the current format bad ?

Reply:
It's just neater to have different endpoints for the different methods.

Reply:
Yea true...